src/Controller/JobController.php

```diff
@@ -0,0 +1,259 @@
+     * gets the job by the Name
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function readJobsByName(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByName($name);
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    $rows
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+  
+    /**
+     * read a job by the substring of the name
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function readJobsByNameLike(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByNameLike($name);
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    $rows
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updates a job from the database
+     *
+     * @param  mixed $id
+     * @param  mixed $request
+     * @return Response
+     */
+    public function updateJob(int $id, Request $request): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $name = $request->get('name');
+            $this->jobValidator->nameIsValid($name);
+            $requestParams = $request->query->all();
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => false,
+                        'rows_updated' => $updateResult
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * delete a job from the database
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $deletedId = null;
+            $jobToDelete = $this->jobsRepository->find($id);
```

Comment:
The find() function from $this->jobsRepository->find($id) will always return one of this:

- JobEntity object if element with $id found in the database
- NULL if no element with id found in the database.
After calling this function always check theif the response is null or not and continue with the code execution only if we have a not null response. Like online 56. Do this for all the places where you call find() function.